### PR TITLE
Add property support to modules and sundry other things

### PR DESCRIPTION
**Pull request by austenmc**
Realized we need a few things while Im working on real modules, like the Weapon Shop. For one, we need a place to store per-module scratch data, so adding property support for Modules.

### CODE REVIEW

**Comment by retropipes** [comment]
Hooray! This change is a GOOD thing.

File: src/Module.php [hunk 1]
```diff
@@ -13,13 +15,13 @@
      * to the database.
      * @param Game $g The game.
```
Comment (Vassyli) [inline src/Module.php:16]: `@param` is missing here.

File: src/Module.php [hunk 2]
```diff
@@ -13,13 +15,13 @@
...
      */
-    public static function onRegister(Game $g);
+    public static function onRegister(Game $g, ModuleModel $module);
 
     /**
      * Lifecycle method called when this module is uninstalled. Use this method
      * to tear down any module-specific additions, like database tables added
      * during registration.
      * @param Game $g The game.
```
Comment (Vassyli) [inline src/Module.php:25]: Here as well.

**Comment by Vassyli** [comment]
Fantastic. Fix documentation and this is good to go!

In the long run, we could think about adding additional annotations for doctrine for properties. The trait is great and all, but $property and $owner still need to be written out every time. But that's very low priority.

**Pull request merged**